### ISSUE TITLE
Defer work in extension activation

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode-pull-request-github#4053]
The GHPRI extension currently takes upwards of 300ms to activate on my machine and @Tyriar's. If we could improve when GHPRI gets activated https://github.com/microsoft/vscode-pull-request-github/issues/4046 we could avoid paying this cost frequently, but perhaps there is work that we can defer to reduce overall extension activation time.

**Comment by Tyriar** [comment]
I notice activate is actually an async function here so those numbers might not reflect what we're thinking:

https://github.com/microsoft/vscode-pull-request-github/blob/main/src/extension.ts#L225

I wonder what the time would be if this:

https://github.com/microsoft/vscode-pull-request-github/blob/8e6b32a23f57daa2edc28beadaab46ff70ef7c6d/src/extension.ts#L244-L255

Was:

```ts
vscode.commands.executeCommand('setContext', 'openDiffOnClick', openDiff).then(() => ...)
```

Also I'm guessing async extension activation does not block other extensions activating?

**Comment by alexr00** [comment]
Found some places to improve. I measured from the start of activation until we're ready to display the tree of a checked out PR.
2.2 s speed up on desktop, for a total time of about 7 s.
170 ms speed up on vscode.dev, for a total time of about 2 s. 

GHPRI is much faster on vscode.dev compared to desktop already and it seems like a combination of `node-fetch` being slow to make the first fetch and shelling out to `git`.
